src/System.Management.Automation/engine/Modules/ScriptAnalysis.cs

```diff
@@ -280,6 +280,7 @@ public override AstVisitAction VisitCommand(CommandAst commandAst)
             string commandName = commandAst.GetCommandName();
             if (commandName is null)
             {
+                // GetCommandName only works if the name is a string constant. GetSafeValueVistor can evaluate some safe dynamic expressions
```

Comment:
GetCommandName? Comment for the method says about scripts, modules, aliases. So I'd like to understand what could be going on here.

Reply:
`GetCommandName` is the method that was run on line 280. It failed to get the name because it only handles constants but there's also this `GetSafeValueVisitor` that can handle a bit more dynamic expressions but still be safe. When this visitor sees a scriptblock expression, it tries to create a new scriptblock using the text but that fails when there's any parse errors in the script text. This is why: `& {(New-Guid).<Tab>` fails, but `& {(New-Guid).N<Tab>` does not, because the N removes the parser error from the missing name after the member access token.

Reply:
Thanks! I see GetSafeValue() process StringConstantExpressionAst too so we could not call commandAst.GetCommandName(). Perhaps it was added as fast way vs visitor.